cmd/nri-ecs_test.go

```diff
@@ -43,21 +40,62 @@ func TestIntegrationPublish(t *testing.T) {
 		os.Stdout = stdout
 	}()
 
-	main()
+	cases := map[string]struct {
+		response       string
+		endpointEnvVar string
+	}{
+		"from_v3_metadata_endpoint": {
+			response:       v3,
+			endpointEnvVar: metadata.ContainerMetadataEnvVar,
+		},
+		"from_v4_metadata_endpoint": {
+			response:       v4,
+			endpointEnvVar: metadata.ContainerMetadataV4EnvVar,
+		},
+	}
+	for testCaseName, testData := range cases {
+		testData := testData
 
-	// Read the integration output from the captured stdout
-	b := make([]byte, 1024)
-	n, err := readerOut.Read(b)
-	if err != nil {
-		t.Fatal(err)
+		t.Run(testCaseName, func(t *testing.T) {
+			ts := metadataServer(t, testData.response)
+			t.Setenv(testData.endpointEnvVar, ts.URL)
+
+			ecsIntegration, _ := integration.New("com.newrelic.ecs", "0.0.0")
+
+			assert.NoError(t, main.Run(ecsIntegration, args))
+
+			// Read the integration output from the captured stdout
+			b := make([]byte, 1024)
+			n, err := readerOut.Read(b)
```

Comment:
Instead of stead stdout we could as well use the [WriterOption](https://github.com/newrelic/infra-integrations-sdk/blob/1f48f6c9407a8be59e2494cf7f426563420a46f1/integration/options.go#L14) when creating the integration

Reply:
look nice! i will do that and remove more code of the integration :)